streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -298,6 +298,7 @@ public boolean isRunning() {
     private volatile ThreadMetadata threadMetadata;
     private StreamThread.StateListener stateListener;
     private final Optional<String> getGroupInstanceID;
+    private final String threadIdSuffix; // shortened version of the threadId: {processUUID}-StreamThread-{threadIdx}
```

Comment:
This isn't exactly a suffix. If users supply their own client id, we just use it directly:
```
        final String userClientId = applicationConfigs.getString(StreamsConfig.CLIENT_ID_CONFIG);
        final String applicationId = applicationConfigs.getString(StreamsConfig.APPLICATION_ID_CONFIG);
        if (userClientId.length() <= 0) {
            clientId = applicationId + "-" + processId;
        } else {
            clientId = userClientId;
        }
```

It seems like what you're really after is either a thread id that's independent of the application id (in which case, why?) or a thread id that's independent of the user-configured client id (also, why?)

Reply:
Ah shoot, good point :/ This was my attempt at a compromise between staying relatively short, and containing enough info to actually be useful. Maybe I'm being overly paranoid, but we have twice had to revert the commit which added the `reason` message to `#enforceRebalance` due to perf issues related to strings/string length -- and I've seen some VERY long application/client ids.

I'm not sure how long is too long, but if you look into the underlying consumer/coordinator methods, everywhere else we pass in a "full reason" for logging plus a "short reason" for actually embedding in the request, due to said perf regression with longer strings. 

But maybe we don't actually need the client id here after all? I'm not sure how this is ultimately used but I would hope the broker would log the client id for a given `reason` string in the JoinGroup request..I'm just going to remove this part for now and if we do find it would be useful, I can do a separate PR.